src/with-theme.js

```diff
@@ -0,0 +1,38 @@
+import React, {Component} from 'react'
+
+import {CHANNEL} from './theme-provider'
+import {PropTypes} from './react-compat'
+
+export default function withTheme(ComponentToTheme) {
+  class ThemedComponent extends Component {
+    state = {theme: {}}
+    setTheme = theme => this.setState({theme})
+
+    componentWillMount() {
+      if (this.context[CHANNEL]) {
+        this.setState({theme: this.context[CHANNEL].getState()})
+      }
+    }
+
+    componentDidMount() {
+      if (this.context[CHANNEL]) {
```

Comment:
`withTheme()` is supposed to be used inside a `<ThemeProvider />`.
Shouldn't we throw an error here if `!this.context[CHANNEL]` ?

This would warn the user before he realize he does not have access to `theme` in children glamorous comps.